plutus-core/plutus-core/src/PlutusCore/Data.hs

```diff
@@ -157,10 +253,7 @@ decodeConstr = CBOR.decodeTag64 >>= \case
   t -> fail ("Unrecognized tag " ++ show t)
   where
   decodeConstrExtended = do
-    lenOrIndef <- CBOR.decodeListLenOrIndef
+    CBOR.decodeListLenOf 2
```

Comment:
This will reject lists that use an indefinite encoding.

Reply:
Ah thanks. Not sure what the nicest way to decode a definite-or-indefinite list, but where I really want two elements...

Reply:
In ledger we have a combinator that does this. Lemme see..

Reply:
This is it:
  https://github.com/input-output-hk/cardano-ledger-specs/blob/master/semantics/executable-spec/src/Data/Coders.hs#L146